### ISSUE TITLE
FOSS licensing clarification (in FLA) that dual-licensing is allowed

### ISSUE THREAD

**Issue by silverhook** [contributoragreements/ca-cla-chooser#63]
After confirming with several original authors of the FLA, it seems the FLA was intended to require (through what is currently _§4. License obligations by Us_) the fiduciary/trustee to release the contributed code (also) as FOSS, allowing for dual-licensing even with a proprietary license.

The way that section is written though, it reads as if it requires the contributed code to _only_ be released as FOSS. This does not seem to be what was originally intended.

I propose to change the wording to match what it seems many of us dealing with the FLA for years understood was the case.

What absolutely needs to stay is a hard requirement that _any_ code that is released _needs_ to be released as FOSS as well. E.g. if a contribution was included in a proprietary version, it needs to be included in a FOSS version too.

**Comment by silverhook** [comment]
IIRC, @catharinamaracke, @step21 and I agreed a few months ago that this would indeed make sense.

I will prepare a PR proposal.

**Comment by silverhook** [comment]
Here’s my idea (also fixes part of #65):

# Option 1: any FOSS license

> We agree to (sub)license the Contribution or any Materials containing, based on or derived from your Contribution under the terms of any licenses that are free and open, following respective commonly accepted definitions. For clarification, should we make available any Material under a non-free or non-open license, we must make a version of such Material available under a free and open license too.
> 
> For example, if the Contribution and/or Materials are:
> 
> * software: licenses that are both [Free Software](https://www.gnu.org/philosophy/free-sw.html), as defined by the Free Software Foundation, and [Open Source](https://opensource.org/osd), as defined by the Open Source Initiative;
> * hardware: licenses that are [Open Source Hardware](https://www.oshwa.org/definition/), as defined by the Open Source Hardware Association;
> * content: licenses that are both [Free Culture Licenses](https://freedomdefined.org/Definition), as defined by Freedom Defined, and [Open Licenses](https://opendefinition.org/od/)”, as defined by the Open Knowledge Foundation.

# Option 2: taxatively selected FOSS license(s)

adds the following (before the “For example” paragraph):

> More specifically and in strict accordance with the above paragraph, we agree to (sub)license the Contribution or any Materials containing, based on or derived from the Contribution only under the terms of the following license(s) <span id="tmp-licenses-fsfe">[LIST_OF_LICENSES]</span> (including any right to adopt any future version of a license if permitted), whereby any non-free or non-open licenses need to be listed as well.

# Option 3: external licensing policy

adds the following (before the “For example” paragraph):

> More specifically and in strict accordance with the above paragraph, we agree to (sub)license the Contribution or any Materials containing, based on or derived from the Contribution only in accordance with our licensing policy available at: <span id="tmp-license-policy-location">[location of licensing policy], where any intent of applying non-free or non-open licenses by Us needs to be described as well.

**Comment by catharinamaracke** [comment]
Looks good to me - are we abandoning the option to license under “any" license (including dual licensing or just proprietary licensing) without any external licensing or IP policy?

Catharina

**Comment by silverhook** [comment]
> are we abandoning the option to license under “any" license (including dual licensing or just proprietary licensing) without any external licensing or IP policy?

I don’t think we need to. Option 1 (in its new version) should still work IMHO.

It is questionable how many people are OK with with the following, if they truly grapsed it:
- any FOSS license (really any, you sure?)
- and the option to have it under any proprietary license
- and both may change at any point

…but there’s bound to be some projects and contributors who’re fine with that.

**Comment by silverhook** [comment]
See [here](https://github.com/contributoragreements/ca-cla-chooser/pull/94/files#diff-0700763486b155a926782de21f0eff675c5311a1a758f198c3bf84bdd16d03aaR88-R109) for how this suggestion would looks like in the text itself.

**Comment by silverhook** [comment]
We decided to keep it simple and only to FOSS, what was included is the following:

> For clarification, should we make available any Material under a non-free or non-open license, we must make a version of such Material available under a free and open license too.

So the following got removed (or changed) from the draft, and I am logging it here for future reference:

```markdown
<!--TODO: use when you have expanded the license grants to cover also data base rights, masks, hardware topology etc.

For example, if the Contribution and/or Materials are:

* software: licenses that are both [Free Software](https://www.gnu.org/philosophy/free-sw.html), as defined by the Free Software Foundation, and [Open Source](https://opensource.org/osd), as defined by the Open Source Initiative;
* hardware: licenses that are [Open Source Hardware](https://www.oshwa.org/definition/), as defined by the Open Source Hardware Association;
* content: licenses that are both [Free Culture Licenses](https://freedomdefined.org/Definition), as defined by Freedom Defined, and [Open Licenses](https://opendefinition.org/od/)”, as defined by the Open Knowledge Foundation.
-->
```

**Comment by step21** [comment]
Implemented. https://github.com/contributoragreements/ca-cla-chooser/blob/dcdc34367bf8006502755c79de029adb6d2f443d/agreement-template-unified.html#L127
